gnosis/safe-react
# Fix trailing slashes breaking app

## What it solves
Resolves #2905

## How this PR fixes it
Sidebar subsections and breadcrumbs (section that includes currency selector) uses the `url` returned from `useRouteMatch`. When matching a route with a trailing slash, the matched `url` includes a trailing slash and therefore no subsection/breadcrumb header matches.

All trailing slashes are removed on all routes via the addition of a new slugged route (`/:url`) that exists only when the `pathname` ends with `/`.

Extra: legacy route redirection had seemingly not been included and L2-UX had passed testing by cached redirects. Nonsense redirection to the welcome page was also remove (see code comment [here](https://github.com/gnosis/safe-react/pull/2907/files#r737434403)).

## How to test it
Open an assets/settings subsection and add a trailing slash to the URL. The sidebar should be open correctly and the breadcrumbs should appear at the top, as well as the currency selector.

Extra: legacy route redirection `/app/#/safes/0x...` => `/app/network:0x...` should work in cache-empty browser. This should be tested in Incognito/Private Browsing so as to have an empty cache.
